services/asymkey/deploy_key.go

```diff
@@ -19,7 +79,19 @@ func DeleteDeployKey(ctx context.Context, doer *user_model.User, id int64) error
 	}
 	defer committer.Close()
 
-	if err := models.DeleteDeployKey(dbCtx, doer, id); err != nil {
+	key, err := asymkey_model.GetDeployKeyByID(ctx, id)
+	if err != nil {
+		if asymkey_model.IsErrDeployKeyNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("GetDeployKeyByID: %w", err)
+	}
+
+	if err := checkDeployPerm(ctx, doer, key.RepoID, key.ID); err != nil {
```

Comment:
It looks like the permission is checked outside of `DeleteDeployKey`. Do we need to check the permissions again using `checkDeployPerm`?

Reply:
The function removed and only check whether the key belongs to the repository.